src/Umbraco.Core/Services/AuditService.cs

```diff
@@ -107,5 +111,72 @@ public IEnumerable<IAuditItem> GetPagedItemsByUser(int userId, long pageIndex, i
                 return repository.GetPagedResultsByQuery(query, pageIndex, pageSize, out totalRecords, orderDirection, auditTypeFilter, customFilter);
             }
         }
+
+        /// <inheritdoc />
+        public IAuditEntry Write(int performingUserId, string perfomingDetails, string performingIp, DateTime eventDate, int affectedUserId, string affectedDetails, string eventType, string eventDetails)
```

Comment:
Should probably put some null checks in place here. I'll add that now. But question, @zpqrtbnk  - should we validate the `eventType` to ensure that it's consistent for whoever is using it? so that it must contain at least one forward slash? and maybe ensure there's no spaces or special chars?

Reply:
eventType totally unspecified at that point - basically free-form text - no clear idea what it's supposed to be - but yea, could at least ensure it has a "path form" ie "foo/bar/nil" - adding validation + a bit of details in the doc.